src/Diagnostics/DiagnosticsMachine/atmos_diagnostic_funs.jl

```diff
@@ -0,0 +1,174 @@
+using ..Atmos
+using ..TurbulenceClosures
+using ..TurbulenceConvection
+
+# Method definitions for diagnostics collection for all the components
+# of `AtmosModel`.
+
+dv_PointwiseDiagnostic(
+    ::AtmosConfigType,
+    ::PointwiseDiagnostic,
+    ::MoistureModel,
```

Comment:
```suggestion
const submodels = Union{
    MoistureModel,
    PrecipitationModel,
    RadiationModel,
    TracerModel,
    TurbulenceClosureModel,
    TurbulenceConvectionModel,
}
dv_PointwiseDiagnostic(::AtmosConfigType,
    ::PointwiseDiagnostic,
    ::submodels,
    ::AtmosModel,
    ::States,
    ::AbstractFloat,
    ::Dict{Symbol, Any},
) = 0
dv_HorizontalAverage(::AtmosConfigType,
    ::HorizontalAverage,
    ::submodels,
    ::AtmosModel,
    ::States,
    ::AbstractFloat,
    ::Dict{Symbol, Any},
) = 0
dv_ScalarDiagnostic(::AtmosConfigType,
    ::ScalarDiagnostic,
    ::submodels,
    ::AtmosModel,
    ::States,
    ::AbstractFloat,
    ::Dict{Symbol, Any},
) = 0
```
Would this be better? Or is the current version protecting against unwanted calls? We could also define `submodels_pointwise`, etc. to reduce the boilerplate a bit.

Reply:
I like your way. Will incorporate.